data-azure-cosmos/src/main/java/io/micronaut/data/cosmos/operations/DefaultCosmosRepositoryOperations.java

```diff
@@ -130,35 +145,31 @@ protected DefaultCosmosRepositoryOperations(List<MediaTypeCodec> codecs,
         this.database = initDatabase(configuration);
     }
 
-    private CosmosDatabase initDatabase(CosmosDatabaseConfiguration configuration) {
-        CosmosDatabaseResponse databaseResponse;
-        ThroughputProperties throughputProperties = createThroughputProperties(configuration);
-        if (throughputProperties == null) {
-            databaseResponse = cosmosClient.createDatabaseIfNotExists(configuration.getDatabaseName());
-        } else {
-            databaseResponse = cosmosClient.createDatabaseIfNotExists(configuration.getDatabaseName(), throughputProperties);
-        }
-        return cosmosClient.getDatabase(databaseResponse.getProperties().getId());
-    }
-
-    private ThroughputProperties createThroughputProperties(CosmosDatabaseConfiguration configuration) {
-        // Throughput properties for the database
-        if (configuration.getThroughputRequestUnits() != null) {
-            if (configuration.isThroughputAutoScale()) {
-                return ThroughputProperties.createAutoscaledThroughput(configuration.getThroughputRequestUnits());
-            } else {
-                return ThroughputProperties.createManualThroughput(configuration.getThroughputRequestUnits());
-            }
-        }
-        return null;
-    }
-
     @Override
     public <T> T findOne(Class<T> type, Serializable id) {
         RuntimePersistentEntity<T> persistentEntity = runtimeEntityRegistry.getEntity(type);
         CosmosContainer container = getContainer(persistentEntity);
-        CosmosItemResponse<ObjectNode> response = container.readItem(id.toString(), PartitionKey.NONE, new CosmosItemRequestOptions(), ObjectNode.class);
-        return deserializeFromTree(response.getItem(), Argument.of(type));
+        try {
+            final SqlParameter param = new SqlParameter("@ROOT_ID", getStringIdValue(id));
+            final SqlQuerySpec querySpec = new SqlQuerySpec(FIND_ONE_DEFAULT_QUERY, param);
+            logQuery(querySpec, Arrays.asList(param));
+            final CosmosQueryRequestOptions options = new CosmosQueryRequestOptions();
+            CosmosPagedIterable<ObjectNode> result = container.queryItems(querySpec, options, ObjectNode.class);
```

Comment:
findById cannot work if there is PartitionKey defined on the container and partition key value not supplied. Since this operations method does not allow passing PartitionKey then we need to use query like this.